MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -216,67 +217,234 @@ namespace WPEFramework {
+                LOGINFO("Invoke status : %d",status);
+                if (status > 0) {
+                    LOGINFO("%s call failed %d", callsign.c_str(), status);
+#if defined(BUILD_ENABLE_NOT_ACTIVATED)
+                    ret_status = "not-activated";
+#else
+                    ret_status = "activation-connect";
+#endif
+                    LOGINFO("Setting Default to [%s]",ret_status.c_str());
+                } else if (joGetResult.HasLabel("status")) {
+                    ret_status = joGetResult["status"].String();
+                    LOGINFO("Activation Value [%s]",ret_status.c_str());
+                }
+                else {
+                    LOGINFO("Defaulting to not-activated/activation-connect");
+#if defined(BUILD_ENABLE_NOT_ACTIVATED)
+                    ret_status = "not-activated";
+#else
+                    ret_status = "activation-connect";
+#endif
+                }
+
+                return ret_status;
+            }
+
+            LOGINFO("thunder client failed");
+            return ret_status;
+        }
+
+        bool MaintenanceManager::isDeviceActivated(bool &skipFirmwareCheck)
+        {
+            /* activation-connect, activation ready, not-activated - execute all except DIFD
+             * activation disconnect - dont run maintenance
+             * activated - run normal */
+            bool ret_result=false;
+            string activationStatus;
+            Auth_activation_status_t result;
+            const std::unordered_map<std::string,std::function<void()>> act{
+                {"activation-connect",   [&](){ result = ACTIVATION_CONNECT; }},
+                    {"activation-ready",   [&](){ result = ACTIVATION_READY; }},
+                    {"not-activated",   [&](){ result = NOT_ACTIVATED; }},
+                    {"activation-disconnect", [&](){ result = ACTIVATION_DISCONNECT; }},
+                    {"activated", [&](){ result = ACTIVATED; }},
+            };
+
+            activationStatus = getActivatedStatus();
+            LOGINFO("activation status : [ %s ]",activationStatus.c_str());
+            const auto end = act.end();
+            auto search = act.find(activationStatus);
+            if ( search != end ){
+                search->second();
+            }
+            else{
+#if defined(BUILD_ENABLE_NOT_ACTIVATED)
+                result = NOT_ACTIVATED;
+                LOGINFO("result: not-activated");
+#else
+                result = ACTIVATION_CONNECT;
+                LOGINFO("result: activation-connect");
+#endif
+            }
+
+            switch(result){
+                case ACTIVATED:
+                    ret_result = true;
+                    break;
+                case ACTIVATION_DISCONNECT:
+                    ret_result = false;
+                    break;
+                default:
+                    ret_result = true;
```

Comment:
checking why is ret_result set to true in the default case. Shouldn't it be true only for ACTIVATED?

Reply:
modified.